src/components/WindowViewer.js

```diff
@@ -26,14 +26,18 @@ export class WindowViewer extends Component {
    * Request the initial canvas on mount
    */
   componentDidMount() {
-    const { fetchInfoResponse } = this.props;
+    const { fetchInfoResponse, fetchAnnotation } = this.props;
 
     if (!this.infoResponseIsInStore()) {
       this.currentCanvases().forEach((canvas) => {
-        const { imageInformationUri } = new ManifestoCanvas(canvas);
+        const manifestoCanvas = new ManifestoCanvas(canvas);
```

Comment:
I think it made some sense as a first pass to include so much of this request logic in the OSD component early on, but I wonder if we can think about how to avoid using lifecycle methods for this. As this gets more complex, and we need cancellations/editing/etc., this is going to become very difficult to debug. Can we hold off on the component side of this until we make time to think through this?

The general condition I'm trying to avoid is requests being initiated from inside components for any reason. This means they exist outside of our state management system in a way that is invisible and now has to be explicitly managed.

Reply:
In this case, for example, we know in the state that a given window is focusing a given canvas. So initiating these requests can be done in actions (namely on next/previous/setcanvas/addWindow, etc.), using our existing pattern. If that starts to get too complex, then perhaps we should consider using @bodo22's suggestion of refactoring to sagas (https://github.com/ProjectMirador/mirador/pull/1667), and/or discuss it as an architectural pattern in https://github.com/ProjectMirador/mirador/issues/1620.

Reply:
Maybe we can jump on a call to discuss this. My main concern is holding up progress on additional annotation tickets to implement a refactor/new architectural pattern that we aren't yet using.